tests/hikari/impl/test_interaction_server.py

```diff
@@ -123,6 +123,19 @@ def test_is_alive_property_when_active(self, mock_interaction_server):
 
         assert mock_interaction_server.is_alive is True
 
+    @pytest.mark.asyncio()
+    async def test___fetch_public_key_when_lock_is_None(self, mock_interaction_server, mock_rest_client):
+        mock_lock = mock.AsyncMock()
```

Comment:
Lock should already be None in the other tests, if you want to better test locking behaviour you might want to gather like 10 calls to an instance of interaction server which was built in that function's scope to ensure to ensure that the locking behaviour persists (where there'd be no need to explicitly set it to None )

Reply:
There are different test cases for this function and I don't think having them do the same call 10 times would be useful

Reply:
So testing that it doesn't overwrite the lock essentially making it useless isn't a useful test case? But testing that it sets the lock which if it didn't it'd either make a lock each time falling under what I just described but not under this test case or erroring on trying to use None as a lock.

Reply:
This is somehow tested by the use of different tests, but I get what you mean, will change it :+1: